src/main.cpp

```diff
@@ -186,100 +208,192 @@ void addPackOptions(po::options_description& desc)
+        ("bounding-box,b", po::cvRect2d_value()->value_name("W S E N"),
+            "Training area bounding box (optional). Must specify four "
             "coordinates: west longitude, south latitude, east longitude, and north latitude. i.e. "
-            "--bounding-box -84.06163 37.22197 -84.038803 37.240162")
-        ("image-type,i", po::value<string>(), "Image type. i.e. jpg")
-        ("color-mode,c", po::value<string>(), "Color mode. Model parameters will override this if present. Must be one of the following: grayscale, rgb, multiband")
+            "--bounding-box -180 -90 180 90")
+        ("color-mode,c", po::value<string>()->value_name("MODE"),
+            "Color mode. Model parameters will override this if present. Must be one of the following: grayscale, rgb, multiband")
+        ("resolution,r", po::cvSize2d_value()->value_name("WIDTH [HEIGHT]"),
+            "Model pixel resolution (optional).")
         ;
 
-    addPackCaffeOptions(pack);
+    addPackFrameworkOptions(pack);
 
     desc.add(pack);
 }
 
-void addPackCaffeOptions(po::options_description& desc)
+
+
+void addPackFrameworkOptions(po::options_description& desc)
 {
-    po::options_description caffe("Pack Caffe Options");
-    caffe.add_options()
-        ("caffe-model", po::value<string>(), "Caffe model file name.")
-        ("caffe-trained", po::value<string>(), "Caffe trained file name.")
-        ("caffe-mean", po::value<string>(), "Caffe mean file name.");
+    for(const auto& type : classification::ModelIdentifier::types()) {
+        const auto& identifier = *classification::ModelIdentifier::find(type);
+
+        string title("Pack ");
+        title += identifier.prettyType();
+        title += " Options";
 
-    desc.add(caffe);
+        po::options_description framework(title.c_str());
+
+        auto multiDesc = string("Set multiple ") + identifier.prettyType() + " options.";
+
+        framework.add(boost::make_shared<po::option_description>(
+            type.c_str(),
+            po::value<vector<string>>()
+                ->value_name("NAME VALUE [NAME VALUE ...]")
+                ->multitoken(),
+            multiDesc.c_str()));
+
+        // Add descriptions for model items
+        for(const auto& item : identifier.itemDescriptions()) {
+            framework.add(createFrameworkOption(item, identifier.type(), "PATH", " file name"));
+        }
+
+        // Add descriptions for model options
+        for(const auto& item : identifier.optionDescriptions()) {
+            framework.add(createFrameworkOption(item, identifier.type(), "VALUE", ""));
+        }
+
+        desc.add(framework);
+    }
+}
+
+boost::shared_ptr<po::option_description> createFrameworkOption(classification::ItemDescription item, const char* type, const char* name, const char* ending)
+{
+    string option(type);
+    option += "-";
+    option += item.name;
+
+    auto description = item.description + ending;
```

Comment:
Also, it's not that simple. Take the `--caffe-output-layer` argument, there is no "default" per se. There's a output layer deduction  algorithm.